src/components/common/Actions.vue

```diff
@@ -186,16 +186,23 @@ export default class Actions extends Mixins(DateMixin, FilingTemplateMixin, Lega
     this.setIsFilingPaying(false)
 
     const paymentToken = filingComplete?.header?.paymentToken
+    const paymentCompleted = filingComplete.header?.paymentStatusCode === 'COMPLETED'
     if (paymentToken) {
       // redirect to pay and return to the dashboard
       const authUrl = sessionStorage.getItem('AUTH_URL')
       const dashboardUrl = sessionStorage.getItem('DASHBOARD_URL')
+      // const nrNumber = this.$route.query.nrNumber as string
       const returnUrl = encodeURIComponent(dashboardUrl + this.getTempId)
       const payUrl = authUrl + 'makepayment/' + paymentToken + '/' + returnUrl
 
       // assume Pay URL is always reachable
       // otherwise user will have to retry payment later
-      window.location.assign(payUrl)
+      if (!paymentCompleted) {
+        window.location.assign(payUrl)
```

Comment:
Suggest you move "const returnUrl" and "const payUrl" into this if-block (as they don't need to be computed is the payment is completed).